vuu-ui/packages/vuu-layout/test/layout-persistence/RemoteLayoutPersistenceManager.test.ts

```diff
@@ -0,0 +1,311 @@
+
+      expectPromiseRejectsWithError(
+        () => persistence.createLayout(metadata[0], layout),
+        errorMessage
+      );
+    });
+
+    it("rejects with error when metadata in response is falsey", () => {
+      const fetchResponse = {
+        json: () => new Promise((resolve) => resolve({})),
+        ok: true,
+      };
+
+      mockFetch.mockResolvedValue(fetchResponse);
+
+      expectPromiseRejectsWithError(
+        () => persistence.createLayout(metadata[0], layout),
+        "invalid metadata"
+      );
+    });
+
+    it("rejects with error when fetch rejects", () => {
+      mockFetch.mockRejectedValue(fetchError);
+
+      expectPromiseRejectsWithError(
+        () => persistence.createLayout(metadata[0], layout),
+        fetchError.message
+      );
+    });
+  });
+
+  describe("updateLayout", () => {
+    it("resolves when fetch resolves and response is ok", () => {
+      const fetchResponse = {
+        ok: true,
+      };
+
+      mockFetch.mockResolvedValue(fetchResponse);
+
+      const result = persistence.updateLayout(uniqueId, metadata[0], layout);
+
+      expect(result).resolves.toBe(undefined);
+    });
+
+    it("rejects with error when response is not ok", () => {
+      const errorMessage = "Not Found";
+
+      const fetchResponse = {
+        ok: false,
+        statusText: errorMessage,
+      };
+
+      mockFetch.mockResolvedValue(fetchResponse);
+
+      expectPromiseRejectsWithError(
+        () => persistence.updateLayout(uniqueId, metadata[0], layout),
+        errorMessage
+      );
+    });
+
+    it("rejects with error when fetch rejects", () => {
+      mockFetch.mockRejectedValue(fetchError);
+
+      expectPromiseRejectsWithError(
+        () => persistence.updateLayout(uniqueId, metadata[0], layout),
+        fetchError.message
+      );
+    });
+  });
+
+  describe("deleteLayout", () => {
+    it("resolves when fetch resolves and response is ok", () => {
+      const fetchResponse = {
+        ok: true,
+      };
+
+      mockFetch.mockResolvedValue(fetchResponse);
+
+      const result = persistence.deleteLayout(uniqueId);
+
+      expect(result).resolves.toBe(undefined);
+    });
+
+    it("rejects with error when response is not ok", () => {
+      const errorMessage = "Not Found";
+
+      const fetchResponse = {
+        ok: false,
+        statusText: errorMessage,
+      };
+
+      mockFetch.mockResolvedValue(fetchResponse);
+
+      expectPromiseRejectsWithError(
+        () => persistence.deleteLayout(uniqueId),
+        errorMessage
+      );
+    });
+
+    it("rejects with error when fetch rejects", () => {
+      mockFetch.mockRejectedValue(fetchError);
+
+      expectPromiseRejectsWithError(
+        () => persistence.deleteLayout(uniqueId),
+        fetchError.message
+      );
+    });
+  });
+
+  describe("loadMetadata", () => {
+    it("resolves with array of metadata when response is ok", () => {
+      const persistence = new RemoteLayoutPersistenceManager();
```

Comment:
Do we need to instantiate a separate persistence manager for each of the `loadMetadata`/`loadLayout` tests, or can we keep using the one in the outer scope?

Reply:
we don't need a new instance here. Not sure why that's there. I will change

Reply:
removed it! Thanks Peter :)